easybuild/tools/job/gc3pie.py

```diff
@@ -197,7 +197,9 @@ def complete(self):
         self._engine.add(self.jobs)
 
         # in case you want to select a specific resource, call
-        self._engine.select_resource(build_option('job_target_resource'))
+        target_resource = build_option('job_target_resource')
+        if target_resource:
+            self._engine.select_resource(target_resource)
```

Comment:
Can you check that this returns a number > 0 ?  If the return value is 0, then no resources have been selected and GC3Pie will error out later.